docs/why.md

```diff
@@ -1,37 +0,0 @@
-# Why Cirrus CI
```

Comment:
I think you deleted this by accident

Reply:
Nope. Just cleaning up. It wasn't in the navigation for a long time because it repeated many points from `features.md` and not as relevant any more. 🤷‍♂

Reply:
Ah.